### DISCUSSION TITLE
Do not treat "Unable to detect disks suitable for backup" as an error?

### DISCUSSION THREAD

**Discussion by dvitek** [abbbi/virtnbdbackup#230] [Ideas]
I am trying to automate the bulk back up of a large number of VMs.  Some of them legitimately do not have disks to back up, but it is still desirable that the xml metadata be backed up.  I use the exit code of virtndbbackup to determine whether it has succeeded.  It exits with failure in cases where a VM has no disks to back up.

I understand that this is reasonable default behavior since it would usually not be intentional to back up only the xml.  However, in my case the backup operation is automatically backing up every persistent VM across many hypervisors, and some of the VMs are expected to be of this form.  It will be somewhat painful to get humans to consistently flag these VMs where they do not expect a disk backup.

While I recognize I could write code to either attempt to detect whether disks that are subject to backup exist before invoking virtnbdbackup, or attempt to screen scrape the output for this particular error message and the absence of any other errors, both of those options seem fragile in comparison to letting virtnbdbackup's existing logic make the determination.

How would you feel about something like `--allow-no-suitable-disks`?  I assume there is no existing way to whitelist specific errors?

**Comment by abbbi** [comment]
two possible solutions here:

1) use another exit code than 1 (maybe; 3?, 1 and 2 are already used)
2) re-arrange the code to issue an warning instead of an error, so utility would exit with exit code 2 if --strict option is set but that would mean to stuff the code that is detecting warnings into a common function to be used earlier in the app.

solution 2 requires a bit more work, solution 1 might make sense, too.
